app/extensions/__init__.py

```diff
@@ -142,6 +142,25 @@ def process(value):
 
         return process(value)
 
+    def compare_values(self, x, y):
+        # This method is used to determine whether a field has changed.
+        #
+        # This is a problem for lists and dicts because if the user edits the
+        # list or dict in place, "x" and "y" are going to be the same and we
+        # can't determine whether the field has changed.
+        #
+        # For example, if self.jobs was {}, then we do:
+        # self.jobs['job_id'] = {'some': 'stuff'}
+        #
+        # compare_values is going to get {'job_id': {'some': 'stuff'}} twice
+        # because sqlalchemy is unable to get back the previous value of
+        # self.jobs which was {}
+        #
+        # So we can't determine whether the field has actually changed.  If we
+        # return True here, the object does not get saved, so we're going to
+        # just always return False
```

Comment:
curious if the implication here is that we are going to "overly" report this as changed?  (e.g. return `False` even when `x` and `y` have not actually changed.)   it seems like the answer has to be "yes", but maybe its not enough to worry about?

Reply:
We can't detect when something is changed, so we'll just have to always say it has changed.

This shouldn't be a problem because we really only write to the database when we do `db.session.merge(...)` or `db.session.add(...)` and when we write those lines we do want to write to the database.